packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -432,34 +421,31 @@ export default function makeKernelKeeper(kernelStorage, kernelSlog) {
    *        be 'endoZipBase64'.
    */
   function addBundle(bundleID, bundle) {
-    const key = bundleIDToKey(bundleID);
-    assert(!kvStore.has(key), 'bundleID already installed');
-    // we repack the object to ensure the DB only holds the known fields
-    const { moduleFormat, endoZipBase64 } = bundle;
-    assert.equal(moduleFormat, 'endoZipBase64');
-    assert.typeof(endoZipBase64, 'string');
-    const value = JSON.stringify({ moduleFormat, endoZipBase64 });
-    kvStore.set(key, value);
+    assert(!bundleStore.hasBundle(bundleID), 'bundleID already installed');
+    bundleStore.addBundle(bundleID, bundle);
   }
 
   /**
    * @param {BundleID} bundleID
    * @returns {boolean}
    */
   function hasBundle(bundleID) {
-    return kvStore.has(bundleIDToKey(bundleID));
+    return bundleStore.hasBundle(bundleID);
   }
 
   /**
    * @param {BundleID} bundleID
    * @returns {EndoZipBase64Bundle | undefined}
    */
   function getBundle(bundleID) {
-    const value = kvStore.get(bundleIDToKey(bundleID));
-    if (value) {
-      return JSON.parse(value);
+    if (bundleStore.hasBundle(bundleID)) {
+      const bundle = bundleStore.getBundle(bundleID);
+      if (bundle.moduleFormat !== 'endoZipBase64')
```

Comment:
you ok with the lack of braces? I am, apparently :)

Reply:
¡Doh!